subprojects/docs/src/docs/release/notes.md

```diff
@@ -355,21 +355,39 @@ To return to the old behavior, explicitly set the classes directory:
 
 Please be aware that this will interfere with the effectiveness of the build cache when using multiple JVM languages in the same source set. Gradle will disable caching for tasks when it detects that multiple tasks create outputs in the same location.
 
+### Detecting test classes for custom `Test` tasks
+
+Before Gradle 4.0, all test classes were compiled into a single output directory. As [described above](#multiple-class-directories-for-a-single-source-set), Gradle now uses separate classes directories for each language in a source set. 
+
+Builds that define custom `Test` tasks may not find the same test classes due to these changes if tests are written in languages other than Java. Deprecation warnings warn about this behavior.
```

Comment:
I think we'll have to explicitly point out that tests are not run if the deprecation message is not fixed.